### ISSUE TITLE
Rename function / find all references doesn't affect inline object literals

### ISSUE THREAD

**Issue by fsoikin** [Microsoft/TypeScript#282]
Check out this code:

```ts
interface I {
	X?: number;
	Y?: string;
}

var i: I;
i.X = 5;

function f() {
	return <I> { X: 1, Y: 'a' };
}
```

([playground link](http://www.typescriptlang.org/Playground/#src=interface%20I%20%7B%0D%0A%09X%3F%3A%20number%3B%0D%0A%09Y%3F%3A%20string%3B%0D%0A%7D%0D%0A%0D%0Avar%20i%3A%20I%3B%0D%0Ai.X%20%3D%205%3B%0D%0A%0D%0Afunction%20f()%20%7B%0D%0A%09return%20%3CI%3E%20%7B%20X%3A%201%2C%20Y%3A%20'a'%20%7D%0D%0A))

Notice how the `I.X` member is referenced two times - once in the `i.X` assignment and once in the object literal inside function `f()`. Now, if I put my caret on the `X` definition and hit F2, the definition itself and the `i.X` assignment get renamed, - but the object literal remains as it was. Worse still, because `X` is an optional property, the compiler doesn't catch the rename at compile time either, and I end up with a good old JavaScript typo situation.

I do realize that it is not always possible for the compiler to know if an object literal is intended to represent an interface or not. For example:

```ts
interface I {
	X?: number;
}
interface J {
	X?: number;
	Y?: string;
}

var i: I = f();
var j: J = f();

function f() {
	return { X: 1 }
}
```

However, I am willing to go an extra mile to help the compiler out here. That's why I put the `<I>` annotation on the object literal in the first example. The problem, however, is that it's impossible for me to lend the compiler a hand, even when I want to.

This is a very real and painful problem when working with property bags, which is a very common JS pattern.

**Comment by RyanCavanaugh** [comment]
Not really anything tricky here, we should just do this

**Comment by mhegazy** [comment]
Looks like this is fixed to me :)

![rename](https://cloud.githubusercontent.com/assets/8000722/7508195/27aad9a2-f431-11e4-803d-7e1ca695409f.gif)
